Kernel/InterruptHandler.cpp

```diff
@@ -0,0 +1,53 @@
+#include <Kernel/InterruptHandler.h>
+#include <Kernel/SharedInterruptHandler.h>
+
+InterruptHandler::InterruptHandler(u8 irq)
+    : m_irq_number(irq)
+{
+    SharedInterruptHandler::get_shared_interrupt_handler(m_irq_number).register_interrupt_handler(*this);
+    disable_irq();
+}
+
+InterruptHandler::~InterruptHandler()
+{
+    disable_irq();
+    SharedInterruptHandler::get_shared_interrupt_handler(m_irq_number).unregister_interrupt_handler(*this);
+}
+
+bool InterruptHandler::is_enabled()
+{
+    return m_enabled;
+}
+
+void InterruptHandler::enable_irq()
+{
+    m_enabled = true;
+}
+
+void InterruptHandler::change_irq_number(u8 irq_number)
+{
+    bool was_enabled = m_enabled;
+    disable_irq();
+    SharedInterruptHandler::get_shared_interrupt_handler(m_irq_number).unregister_interrupt_handler(*this);
+    m_irq_number = irq_number;
+    SharedInterruptHandler::get_shared_interrupt_handler(m_irq_number).register_interrupt_handler(*this);
+    if(was_enabled)
+        enable_irq();
+}
+
+void InterruptHandler::disable_irq()
+{
+    m_enabled = false;
+}
+
+InterruptEnabler::InterruptEnabler(InterruptHandler& handler)
+    : m_handler(handler)
+    , m_was_enabled(m_handler.is_enabled())
+{
+    m_handler.enable_irq();
+}
+InterruptEnabler::~InterruptEnabler()
+{
+    if(!m_was_enabled)
+        m_handler.disable_irq();
+}
```

Comment:
Missing newline at end of file, please configure your editor.

Reply:
I fixed that, thanks for the advice! :)